x-parser.js

```diff
@@ -551,15 +531,24 @@ export class XParser {
     throw new Error(message);
   }
 
-  // This validates a value from our “strings.raw” array passed into our tagged
-  //  template function. It checks to make sure superfluous, JS-y escapes are
-  //  not being used as html (since there are perfectly-valid alternatives).
+  // Character escapes like “\n”, “\u” or ”\x” are a JS-ism. We want developers
+  //  to use HTML here, not JS. You can of course interpolate whatever you want.
+  //  https://w3c.github.io/html-reference/syntax.html#character-encoding
+  //  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_escape
+  // Note that syntax highlighters expect the text after the “html” tag to be
+  //  real HTML. Another reason to reject JS-y unicode is that it won’t be
+  //  interpreted correctly by tooling that expects _html_.
+  // Escapes for the “$”, “\”, and “`” characters are required within a template
+  //  literal — users will need to use “&dollar;”, “&bsol;”, and “&grave;”.
+  // Examples:
+  //  - ok: html`&#8230;`, html`&#x2026;`, html`&mldr;`, html`&hellip;`
+  //  - not ok: html`\nhi\nthere`, html`\x8230`, html`\u2026`, html`\s\t\o\p\ \i\t\.`
   static #validateRawString(rawString) {
-    XParser.#rawJsEscape.lastIndex = 0;
-    if (XParser.#rawJsEscape.test(rawString)) {
```

Comment:
This is the performance gain. Again, small, but non-negligible. A couple microseconds per interpretation. It’s also much easier to quickly grok what this is doing with `rawString.indexOf('\\')`.